src/js/utils/selection-utils.js

```diff
@@ -20,11 +36,20 @@ function comparePosition(selection) {
     headOffset = focusOffset; tailOffset = anchorOffset;
     direction = DIRECTION.BACKWARD;
   } else { // same node
-    headNode = anchorNode;
-    tailNode = focusNode;
-    headOffset = Math.min(anchorOffset, focusOffset);
-    tailOffset = Math.max(anchorOffset, focusOffset);
-    direction = null;
+    headNode = tailNode = anchorNode;
+    if (anchorOffset > focusOffset) {
+      headOffset = focusOffset;
+      tailOffset = anchorOffset;
+      direction = DIRECTION.BACKWARD;
+    } else if (anchorOffset < focusOffset) {
+      headOffset = anchorOffset;
+      tailOffset = focusOffset;
+      direction = DIRECTION.FORWARD;
+    } else {
+      headOffset = anchorOffset;
+      tailOffset = focusOffset;
+      direction = null;
+    }
```

Comment:
More of a stylistic suggestion, to me it's more understandable to dry the differences between the 3 scenarios like so:
```
headNode = tailNode = anchorNode;
headOffset = Math.min(anchorOffset, focusOffset); // head always the smaller number
tailOffset = Math.max(anchorOffset, focusOffset); // tail always the larger number
// direction = null if equal, else forward if focus is > offset, backward otherwise
direction = anchorOffset === focusOffset ? null : (focusOffset > anchorOffset ? DIRECTION.FORWARD : DIRECTION.BACKWARD);
```